pkg/flow/temporary.go

```diff
@@ -238,13 +238,26 @@ func (im *instanceMemory) SetVariables(ctx context.Context, vars []states.Variab
 			return derrors.NewInternalError(errors.New("invalid scope"))
 		}
 
+		// if statements have to be same order
+
 		d := string(v.Data)
-		if len(d) < 3 { // checking for < 3 cause empty data looks like {}
-			_, _, err = im.engine.flow.DeleteVarialbe(ctx, vrefc, vdatac, q, v.Key, v.Data, v.MIMEType, thread)
+		if (v.MIMEType == "text/plain; charset=utf-8" || v.MIMEType == "text/plain" || v.MIMEType == "application/octet-stream") && len(d) == 0 {
```

Comment:
I still think these checks aren't quite right. A plaintext or binary input of `{}` will be deleted, when it shouldn't be. 

1. IF `len(d) == 0`, delete.
2. IF NOT `(v.MIMEType == "text/plain; charset=utf-8" || v.MIMEType == "text/plain" || v.MIMEType == "application/octet-stream")` AND `(d == "{}" || d == "[]" || d == "0" || d == "" || d == "null")`, delete.
3. ELSE, set